### PR TITLE
Sql Import/Export CLI commands and test

### PR DESCRIPTION
**Pull request by nathannfan**
```
az sql db import -h
Command
    az sql db import: Imports a bacpac into an existing database.

Arguments
    --admin-password -p [Required]: The password of the SQL administrator.
    --admin-user -u     [Required]: The name of the SQL administrator.
    --name -n           [Required]: Name of the Azure SQL Database.
    --resource-group -g [Required]: Name of resource group. You can configure the default group
                                    using 'az configure --defaults group=<name>'.
    --server -s         [Required]: Name of the Azure SQL server.
    --storage-key       [Required]: The storage key to use.
    --storage-key-type  [Required]: The type of the storage key to use. Valid values are
                                    StorageAccessKey and SharedAccessKey.  Allowed values:
                                    SharedAccessKey, StorageAccessKey.
    --storage-uri       [Required]: The storage uri to use.
    --auth_type                   : The authentication type - if not specified, will default to SQL.
                                    Allowed values: ADPassword, SQL.
    --type                        : The type of the extension.
```

```
az sql db export -h
Command
    az sql db export: Exports a database to a bacpac.

Arguments
    --admin-password -p [Required]: The password of the SQL administrator.
    --admin-user -u     [Required]: The name of the SQL administrator.
    --name -n           [Required]: Name of the Azure SQL Database.
    --resource-group -g [Required]: Name of resource group. You can configure the default group
                                    using 'az configure --defaults group=<name>'.
    --server -s         [Required]: Name of the Azure SQL server.
    --storage-key       [Required]: The storage key to use.
    --storage-key-type  [Required]: The type of the storage key to use. Valid values are
                                    StorageAccessKey and SharedAccessKey.  Allowed values:
                                    SharedAccessKey, StorageAccessKey.
    --storage-uri       [Required]: The storage uri to use.
    --auth_type                   : The authentication type - if not specified, will default to SQL.
                                    Allowed values: ADPassword, SQL.
```

### CODE REVIEW

**Comment by jaredmoo** [comment]
How does SharedAccessKey work? Do you put the SAS value in for storage key? What happens if you put it on the end of the file uri instead?

File: azure-cli.pyproj [hunk 1]
```diff
@@ -526,11 +526,12 @@
     <Folder Include="command_modules\azure-cli-component\azure\cli\" />
     <Folder Include="command_modules\azure-cli-component\azure\cli\command_modules\" />
     <Folder Include="command_modules\azure-cli-component\azure\cli\command_modules\component\" />
+    <Folder Include="command_modules\azure-cli-component\azure-cli-configure\" />
+    <Folder Include="command_modules\azure-cli-component\azure-cli-configure\azure\" />
+    <Folder Include="command_modules\azure-cli-component\azure-cli-configure\azure\cli\" />
+    <Folder Include="command_modules\azure-cli-component\azure-cli-configure\azure\cli\command_modules\" />
+    <Folder Include="command_modules\azure-cli-component\azure-cli-configure\azure\cli\command_modules\configure\" />
```
Comment (jaredmoo) [inline azure-cli.pyproj]: Why did these changes happen? Maybe these should be reverted?

Comment (nathannfan) [inline azure-cli.pyproj]: Yep, must have happened during a merge.

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/commands.py [hunk 2]
```diff
@@ -99,6 +101,7 @@
                                            'ServersOperations')
 
 with ServiceGroup(__name__, get_sql_servers_operation, server_operations) as s:
+
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/commands.py]: nitpicking: this is not required.

**Comment by jaredmoo** [comment]
`--type` param shouldn't be there, right?

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py [hunk 3]
```diff
@@ -865,3 +866,84 @@ def test_sql_elastic_pools_mgmt(self, resource_group, resource_group_location, s
...
+                 ' --storage-uri {}{}/testbacpac.bacpac'
+                 .format(server, db_name, rg, admin_password, admin_login, key,
+                         storage_endpoint, container))
+
+        # import bacpac to second database
+        self.cmd('sql db import -s {} -n {} -g {} -p {} -u {}'
+                 ' --storage-key {} --storage-key-type StorageAccessKey'
+                 ' --storage-uri {}{}/testbacpac.bacpac'
+                 .format(server, db_name2, rg, admin_password, admin_login, key,
+                         storage_endpoint, container))
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: There aren't any assertion on the last two command execution except the exit code. Shall we add more assertions?

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py [hunk 4]
```diff
@@ -865,3 +866,84 @@ def test_sql_elastic_pools_mgmt(self, resource_group, resource_group_location, s
...
+        location_long_name = 'West US'
+        admin_login = 'admin123'
+        admin_password = 'SecretPassword123'
+        db_name = 'cliautomationdb01'
+        db_name2 = 'cliautomationdb02'
+        container = 'bacpacs'
+
+        firewall_rule_1 = 'allowAllIps'
+        start_ip_address_1 = '0.0.0.0'
+        end_ip_address_1 = '255.255.255.255'
```
Comment (jaredmoo) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: does it work if you just allow azure ips? i.e. start_ip_address = 0.0.0.0, end_up_address = 0.0.0.0

**Comment by jaredmoo** [comment]
Can you make a swagger change to deweirdify this message? You can remove `Valid values are StorageAccessKey and SharedAccessKey.` from the spec.

```
    --storage-key-type  [Required]: The type of the storage key to use. Valid values are
                                    StorageAccessKey and SharedAccessKey.  Allowed values:
                                    SharedAccessKey, StorageAccessKey.
```

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py [hunk 5]
```diff
@@ -865,3 +866,84 @@ def test_sql_elastic_pools_mgmt(self, resource_group, resource_group_location, s
...
+
+        loc_long = location_long_name
+        rg = resource_group
+        sa = storage_account
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: Are these assignment just for the sake of a shorter name?

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py [hunk 6]
```diff
@@ -865,3 +866,84 @@ def test_sql_elastic_pools_mgmt(self, resource_group, resource_group_location, s
...
+    @ResourceGroupPreparer()
+    @SqlServerPreparer()
+    @StorageAccountPreparer()
+    def test_sql_db_import_export_mgmt(self, resource_group, resource_group_location, server, storage_account):
+        location_long_name = 'West US'
+        admin_login = 'admin123'
+        admin_password = 'SecretPassword123'
+        db_name = 'cliautomationdb01'
+        db_name2 = 'cliautomationdb02'
+        container = 'bacpacs'
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: Can we use the `self.create_random_name` for all these names?

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py (hunk 5 above)
Comment (nathannfan) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: Yep

**Comment by jaredmoo** [comment]
Looks great overall, just curious how SAS keys work. Could you add examples for each key type to help doc? Take a look at my help.py changes in https://github.com/Azure/azure-cli/pull/2536 to see how to format examples.

Test for SAS key would be nice to have but not required.

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py (hunk 4 above)
Comment (nathannfan) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: It does

Comment (nathannfan) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: It does

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py (hunk 3 above)
Comment (nathannfan) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: Done

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py (hunk 6 above)
Comment (jaredmoo) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: Why is random preferred? Db names only need to be unique within a server.

Comment (troydai) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: In that case, it should be fine.

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py (hunk 5 above)
Comment (troydai) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: None blocking, but not recommended. Longer variable names improve the readability. Reassign and shorten the names defeat the purpose.

**Comment by nathannfan** [comment]
@troydai Thanks!

**Pull request merged**